tests/test_arrow_dataset.py

```diff
@@ -569,7 +580,7 @@ def test_concatenate_with_indices(self, in_memory):
             )
             dset1, dset2, dset3 = self._to(in_memory, tmp_dir, dset1, dset2, dset3)
             dset1, dset2, dset3 = dset1.select([0, 1, 2]), dset2.select([0, 1, 2]), dset3
-
+            # TODO(QL): make sure concatenation with indices doesn't use RAM
```

Comment:
I guess this will be addressed in another PR...

Reply:
Yes exactly.
I think I'll do one PR for `concatenate_datasets` and one for `filter`